chiselFrontend/src/main/scala/chisel3/Reg.scala

```diff
@@ -178,12 +178,15 @@ object RegInit {
       requireIsChiselType(t, "reg type")
     }
     val reg = t.cloneTypeFull
-    val clock = Builder.forcedClock.ref
-    val reset = Builder.forcedReset.ref
+    val clock = Builder.forcedClock
+    val reset = Builder.forcedReset
+    if (reset.isInstanceOf[AsyncReset] && init.litOption.isEmpty) {
```

Comment:
I assume this an opportunistic check that won't detect non-literal-resets for AsyncResets inferred in FIRRTL? If so, might be worth noting this limitation in the comments.

Reply:
Removed in favor of more sophisticated check in FIRRTL